pkg/healthcheck/healthcheck.go

```diff
@@ -1415,6 +1420,24 @@ func (hc *HealthChecker) allCategories() []*Category {
 			},
 			false,
 		),
+		NewCategory(
+			LinkerdOpaquePortsDefinitionChecks,
+			[]Checker{
+				{
+					description: "opaque ports are properly annotated",
```

Comment:
Since this is a check which validates data plane configuration (i.e. annotations on pods and services in the data plane) rather than anything about the control plane, I think this check belongs in the LinkerdDataPlaneChecks category rather than in a brand new category.  For example, it is conceptually similar to the "data plane service labels are configured correctly" check.  

This will mean that this check will only be run when the user passes the `--proxy` flag to `linkerd check`.